internal/validation/access_strategy.go

```diff
@@ -25,3 +25,24 @@ func CheckForExclusiveAccessStrategy(accessStrategies []*gatewayv1beta1.Authenti
 
 	return nil
 }
+
+// CheckForSecureAndUnsecureAccessStrategies checks if there are secure and unsecure access strategies used at the same time.
+func CheckForSecureAndUnsecureAccessStrategies(accessStrategies []*gatewayv1beta1.Authenticator, attributePath string) []Failure {
```

Comment:
Is there a reason why we didn't add unit tests on package level? Might be confusing for someone that changes the functionality of a method like this.

Reply:
Added unit tests.